sw/device/lib/runtime/otbn.h

```diff
@@ -234,4 +234,14 @@ otbn_result_t otbn_copy_data_from_otbn(otbn_t *ctx, size_t len_bytes,
  */
 otbn_result_t otbn_zero_data_memory(otbn_t *ctx);
 
+/**
+ * Writes a LOG_INFO message with the contents of each 256b DMEM word.
+ *
+ * @param ctx The context object.
+ * @param max_addr The highest address to dump. Set to 0 to output the whole
+ *                 DMEM.
+ * @return The result of the operation.
+ */
+otbn_result_t otbn_dump_dmem(const otbn_t *ctx, uint32_t max_addr);
```

Comment:
Start address might be useful in future if someone wants to dump a particular variable. Up to you. Probably best to have `start_addr` and `max_length` in that case.

Reply:
In my debugging experience I didn't need that so far. All relevant data is at the start of DMEM, and dumping more than just the required variable is very helpful, as the in many cases I need to find a data word in the dump that is slightly misplaced or shifted.

I'd prefer to keep it simple for now, and expand as necessary.